src/root_hits.jl

```diff
@@ -0,0 +1,88 @@
+"""
+    RootHitFile(file::Union{IOStream, AbstractString}; batch_size::Integer=10)
+
+represents a `.root.hits` file, which can be iterated or read to yield events, containg data
+on energy depositions, i.e. hits.
+
+A `RootHitFile` is also a Tables.jl compatible row table of events. `batch_size` determines
+the number of events grouped into partitions when `Tables.partitions` is used. To change the
+default of 10, see `ROOT_HITS_BATCH_SIZE`.
+"""
+struct RootHitFile
+    stream::IOBuffer
+    batch_size::Int
+
+    function RootHitFile(stream::IOStream; batch_size=ROOT_HITS_BATCH_SIZE[])
+        new(IOBuffer(mmap(stream)), batch_size)
+    end
+end
+
+function RootHitFile(path::AbstractString; batch_size=ROOT_HITS_BATCH_SIZE[])
+    if occursin(r".root.hits$", path)
+        return RootHitFile(open(path); batch_size=batch_size)
+    else
+        throw(ArgumentError("$path is not a .root.hits file"))
+    end
+end
+
+const ROOT_HITS_BATCH_SIZE = Ref(10)
+
+const RootHitEventTuple = NamedTuple{
+    (:eventnum, :primcount, :pos, :E, :time, :particleID, :trkID, :trkparentID, :volumeID),
+    Tuple{
+        Int32, Int32, Vector{NTuple{3, Float32}}, Vector{Float32}, Vector{Float32},
```

Comment:
Might be good to use a `StaticArrays.SVector{3}` instead of an `NTuple{3}`, since `SVector` supports LinearAlgebra (we'll want to translate, rotate, etc. hit positions).

Reply:
I had this before. I think I had a phase where I wanted to simplify everything so I threw it out. Probably went too far on my quest for simplification on that point.

This, by the way, is why I don't have a `:32` / `:64` type parameter to specify parsing as 32 bit or 64 bit ints / floats. It's simpler and until I'm convinced anybody actually needs 64 bit integers / precision I might as well save myself the trouble of implementing it. Would be curious if you think it's worth adding the extra complexity.

Reply:
I agree, in a use case like this we should just choose a precision and stick with it. It's different for computational code that may have to handle dual numbers and things like this, but for input, fixed types are usually fine.